bitcoin/dogeblock.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2020 Coinbase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bitcoin
+
+import (
+	"encoding/binary"
+	"io"
+
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/wire"
+)
+
+const (
+	vChainID       = 0x0062
+	vChainIDLsbPos = 16
+	vChainIDMask   = 0xFFFF
+	vAuxPow        = 0x0100
+)
+
+// MerkleBranch defines a merkel branch
+// https://en.bitcoin.it/wiki/Merged_mining_specification#Merkle_Branch
+type MerkleBranch struct {
+	BranchHashes   []*chainhash.Hash
+	BranchSideMask int32
+}
+
+// DogeBlock defines a dogecoin block message.
+// It is used to deliver block and transaction information.
+// https://en.bitcoin.it/wiki/Merged_mining_specification#Aux_proof-of-work_block
+type DogeBlock struct {
+	Header           wire.BlockHeader
+	CoinbaseTx       wire.MsgTx
+	BlockHash        chainhash.Hash
+	CoinbaseBranch   MerkleBranch
+	BlockchainBranch MerkleBranch
+	ParentHeader     wire.BlockHeader
```

Comment:
Can we put this in a separate struct? i.e. `AuxHeader` - we're anyway not going to validate it and most likely even skip over it, because we're reading from a "trusted" fully validating node.